pkg/terraform/exec/plugins/vsphereprivate/resource_vsphereprivate_import_ova.go

```diff
@@ -189,11 +204,45 @@ func findImportOvaParams(client *vim25.Client, datacenter, cluster, datastore, n
 	for _, hostObj := range hosts {
 		foundDatastore := false
 		foundNetwork := false
-		err := hostObj.Properties(ctx, hostObj.Reference(), []string{"network", "datastore", "runtime"}, &hostSystemManagedObject)
+		err := hostObj.Properties(ctx, hostObj.Reference(), []string{"config.product", "network", "datastore", "runtime"}, &hostSystemManagedObject)
 		if err != nil {
 			return nil, err
 		}
 
+		// If HardwareVersion is 13 there is no reason to continue checking
+		// There is a ESXi host that does not support hardware 15.
+		if importOvaParams.HardwareVersion != vmx13 {
+			esxiHostVersion, err := version.NewVersion(hostSystemManagedObject.Config.Product.Version)
+			if err != nil {
+				return nil, err
+			}
+
```

Comment:
Would this be better?
```
importOvaParams.HardwareVersion = vmx13
if esxiHostVersion.Equal(v67) {
	build, err := strconv.Atoi(hostSystemManagedObject.Config.Product.Build)
	if err != nil {
		return nil, err
	}
	// This is the ESXi 6.7 U3 build number
	// Anything less than this version is unsupported with the
	// out-of-tree CSI.
	// https://kb.vmware.com/s/article/2143838
	// https://vsphere-csi-driver.sigs.k8s.io/supported_features_matrix.html
	if build >= Esxi67U3BuildNumber {
		importOvaParams.HardwareVersion = vmx15
	}
} else if esxiHostVersion.GreaterThan(v67) {
	importOvaParams.HardwareVersion = vmx15
}
```

Reply:
@rna-afk thanks for the suggestion, yes so much better with less branches